splitgpg2/__init__.py

```diff
@@ -831,29 +855,36 @@ def parse_sexpr(klass, untrusted_arg):
         return sexpr[0]
 
     @classmethod
-    def _parse_sexpr(klass, untrusted_arg):
+    def _parse_sexpr(klass, untrusted_arg, depth=0, parens=0):
+        if depth > 10:
+            raise ValueError("nesting depth too deep")
         if len(untrusted_arg) == 0:
+            if parens != 0:
+                raise ValueError("Missing close parentheses")
```

Comment:
Already implemented. Sorry for the duplication of work. I didn't push since I hadn't tested the state of my branch yet. Now pushed (not fully tested).